content/blog/2017/05/2017-05-12-kubernetes-journey-on-azure.adoc

```diff
@@ -0,0 +1,397 @@
+---
+layout: post
+title: "A journey to Kubernetes on Azure"
+tags:
+- puppet
+- kubernetes
+- docker
+- azure
+author: olblak
+---
+
+With the
+link:http://localhost:4242/blog/2016/05/18/announcing-azure-partnership/[ongoing migration to Azure],
+I would like to share my thoughts regarding one of the biggest challenges we
+have faced thus far: **orchestrating container infrastructure**. Many of the
+Jenkins project's applications are run as Docker containers, making Kubernetes
+a logical choice as far as running our containers, but it presents its own set
+of challenges. For example, what would the workflow from development to
+production look like?
+
+Before going deeper into the challenges, let's review the requirements we
+started with:
+
+Git::
+  We found it mandatory to keep track of all the infrastructure changes in Git
+  repositories, including secrets, in order to facilitate reviewing,
+  validation, rollback, etc of all infra changes.
+
+Tests::
+  Infrastructure contributors are geographicaly distributed and on different timezone.
+  Getting feedback can take time, so we heavily rely on a lot on tests before
+  any changes can be merged.
+
+Automation::
+  The change submitter is not necessarly the person who will deploy it.
+  Repetitive tasks are error prone and a waste of time.
+  For these reasons, all steps must be automated and stay as simple as possible.
+
+
+A high level overview of our "infrastructure as code" workflow would look like:
+
+
+.Infrastructure as Code Workflow
+----
+  __________       _________       ______________
+  |         |      |        |      |             |
+  | Changes | ---->|  Test  |----->| Deployment  |
+  |_________|      |________|  ^   |_____________|
+                               |
+                        ______________
+                       |             |
+                       | Validation  |
+                       |_____________|
+----
+
+
+
+We identified two possible approaches for implementing our container
+orchestration with Kubernetes:
+
+. The Jenkins Way: Jenkins is triggered by a git commit, runs the tests, and
+  after validation, Jenkins deploys changes into production.
+. The Puppet way: Jenkins is triggered by a git commit, runs the tests, and
```

Comment:
* Inconsistent capitalization of 'way' here and in the line above.
* git should generally be Git?